src/hypothesis/strategies.py

```diff
@@ -857,18 +857,21 @@ def recursive(base, extend, max_leaves=100):
 def permutations(values):
     """Return a strategy which returns permutations of the collection
     "values"."""
+    from hypothesis.internal.conjecture.utils import integer_range
+
     values = list(values)
     if not values:
-        return just(()).map(lambda _: [])
-
-    def build_permutation(swaps):
-        initial = list(values)
-        for i, j in swaps:
-            initial[i], initial[j] = initial[j], initial[i]
-        return initial
-    n = len(values)
-    index = integers(0, n - 1)
-    return lists(tuples(index, index), max_size=n ** 2).map(build_permutation)
+        return builds(list)
```

Comment:
Maybe I’m missing something (in the original if not the new version) – why not use `just([])`?